src/components/signup/Signup.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useEffect, useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { Toaster } from "react-hot-toast";
+import { supabase } from "../../../supabaseClient";
+
+import { SignupType, SignupErrorType } from "./SignupType";
+
+import { useToast, useAuth } from "../../contexts";
+import { Formik, Field, Form, ErrorMessage } from "formik";
+import * as Yup from "yup";
+
+import {
+  emailRegex,
+  wrongEmailMessage,
+  passwordRegex,
+  wrongPasswordMessage,
+} from "../../utils";
+
+const Signup = () => {
+  const { notifySuccess, notifyError } = useToast();
+  const navigate = useNavigate();
+
+  const { currentUser } = useAuth();
+  useEffect(() => {
+    if (currentUser) {
+      window.location.reload();
+      navigate("/dashboard", { replace: true });
+    }
+  }, []);
+
+  const initialValues: SignupType = {
+    username: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  };
+
+  const validationSchema = Yup.object({
+    username: Yup.string().required("username is required"),
+
+    email: Yup.string()
+      .required("Invalid Email Format Email should contain valid email address")
+      .required("Email Address is required")
+      .matches(emailRegex, wrongEmailMessage),
+
+    password: Yup.string()
+      .required("Password is required")
+      .matches(passwordRegex, wrongPasswordMessage),
+
+    confirmPassword: Yup.string()
+      .oneOf([Yup.ref("password")], "Mismatched passwords")
+      .required("Please confirm your password"),
+  });
+
+  const signupFormValidate = async (values: SignupType) => {
+    console.log(values.confirmPassword);
+
+    try {
+      const { user, session, error } = await supabase.auth.signUp({
+        email: values.email,
+        password: values.password,
+      });
+      if (error) throw error;
+      notifySuccess("Account created successfully");
+      navigate("/dashboard", { replace: true });
+      window.location.reload();
```

Comment:
Suggestion: remove this imperative javascript statement. Use declarative way to do it.